### PR TITLE
feat(visibility): Add static sampling to organization tag keys

### PR DESCRIPTION
**Pull request by gggritso**
Related to https://github.com/getsentry/sentry/issues/61999. For organizations who are getting timeouts while fetching organization tags, use a [ClickHouse SAMPLE N](https://clickhouse.com/docs/en/sql-reference/statements/select/sample#sample-n) instead of a sampling rate.

The intent is to give this a try, and then roll out to everyone if it works out well. I'm going with 1,000,000 as a starting point, I'll be tweaking that number soon when I run some experiments.

### CODE REVIEW

File: src/sentry/tagstore/snuba/backend.py [hunk 1]
```diff
@@ -433,6 +434,15 @@ def get_tag_keys_for_projects(
...
         # We want to disable FINAL in the snuba query to reduce load.
         optimize_kwargs = {"turbo": True}
+
+        # Add static sample amount to the query. Turbo will sample at 10% by
+        # default, but organizations with many events still get timeouts. A
+        # static sample creates more consistent performance.
+        organization_id = get_organization_id_from_project_ids(projects)
+        organization = Organization.objects.get_from_cache(id=organization_id)
+        if features.has("organizations:tag-key-sample-n", organization):
+            optimize_kwargs["sample"] = 1_000_000
```
Comment (Zylphrex) [inline src/sentry/tagstore/snuba/backend.py:444]: Thoughts on making this an option so you can tweak the sample size without a deploy?

Comment (gggritso) [reply src/sentry/tagstore/snuba/backend.py:444]: 🤔 that seems like a good idea, although to be honest I'm planning to do research upfront about the right sample number, so I'm not expecting to change this a lot. Maybe I'm being too optimistic?

Comment (Zylphrex) [reply src/sentry/tagstore/snuba/backend.py:444]: I don't expect this to change often, but rather when you do need to change it, it's nice if we didn't have to wait hours to see the change reflected.

One other approach would be to use the admin tool to test various sample rates and just deciding based on that.

**Review by Zylphrex — APPROVED** [review]

File: src/sentry/tagstore/snuba/backend.py (hunk 1 above)
Comment (gggritso) [reply src/sentry/tagstore/snuba/backend.py]: Agreed! Added an option

**Pull request merged**